src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -791,10 +767,8 @@ function buildStats(
         .filter(filterPlugs)
         .filter((socket) =>
           Boolean(
-            socket.plug &&
-              socket.plug.plugItem.plug.plugCategoryHash !== 3347429529 &&
-              socket.plug.plugItem.investmentStats &&
-              socket.plug.plugItem.investmentStats.length
+            idx(socket, (x) => x.plug.plugItem.plug.plugCategoryHash) !== 3347429529 &&
+              idx(socket, (x) => x.plug.plugItem.investmentStats.length)
```

Comment:
Same deal, these could be `socket.plug` like just above.

Reply:
as above & standardized arrow function variables